Client/Assets/Favicons.js

```diff
@@ -2,47 +2,47 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-(function() {
- // These integers should be kept in sync with the IconType raw-values
- var ICON = 0;
- var APPLE = 1;
- var APPLE_PRECOMPOSED = 2;
- var GUESS = 3;
 
- var Favicons = {
-  selectors: { "link[rel~='icon']": ICON,
-              "link[rel='apple-touch-icon']": APPLE,
-              "link[rel='apple-touch-icon-precomposed']": APPLE_PRECOMPOSED },
 
- getAll: function() {
-    var res = {}
-    var foundIcons = false
 
-    for (selector in this.selectors) {
-      var icons = document.querySelectorAll(selector)
-      if (icons.length) {
-        res = {}
-      }
-      for (var i = 0; i < icons.length; i++) {
-        var href = icons[i].href;
-        res[href] = this.selectors[selector];
-          foundIcons = true
+window._FireFox_Favicons = function() {
+    // These integers should be kept in sync with the IconType raw-values
+    var ICON = 0;
+    var APPLE = 1;
+    var APPLE_PRECOMPOSED = 2;
+    var GUESS = 3;
```

Comment:
Nit: Please keep 2-space indentation for JS files. Sorry -- it's confusing! If you aren't already, I suggest using a separate text editor for JS files other than Xcode.